app/models/package_manager/maven.rb

```diff
@@ -192,8 +192,9 @@ def self.retrieve_versions(versions, name)
       versions
         .map do |version|
           one_version_for_name(version, name)
-      rescue Ox::Error, POMNotFound
-        next
+        rescue Ox::Error, POMNotFound
+          StructuredLog.capture("MAVEN_RETRIEVE_VERSION_FAILURE", { version: version, name: name, message: "Version POM not found or valid" })
```

Comment:
🎱 Might be nice to log the url / directory that we attempted to look at here to make it easier for debugging.

Reply:
In order to do this, we'd either have to reconstruct the `MavenUrl` that we generate [here](https://github.com/librariesio/libraries.io/blob/main/app/models/package_manager/maven.rb#L202) which would risk them both getting out of sync, or we would have to pass more arguments between methods, which I think is not necessary. I think that for debugging have this information will likely be sufficient, as it will inform the arguments passed into the method to make the request, so personally I'm inclined to leave this log as it is.